File: uniseek_vue/src/api/resume.ts
import request from './index'

export interface ResumeData {
  id?: number
  userId?: number
  realName?: string
  gender?: number
  birthDate?: string
  education?: string
  school?: string
  skills?: string
  experience?: string
  attachmentUrl?: string
  isPublished?: number
  createTime?: string
  updateTime?: string
}

export interface ResumeSaveParams {
  gender?: number
  birthDate?: string
  education?: string
  school?: string
  skills?: string
  experience?: string
  attachmentUrl?: string
}

/** GET /resume 获取我的简历 */
export const getResume = () =>
  request.get<any, ResumeData>('/resume')

/** PUT /resume 保存简历 */
export const saveResume = (params: ResumeSaveParams) =>
  request.put<any, void>('/resume', params)

/** POST /resume/upload-attachment 上传简历附件 */
export const uploadAttachment = (file: File) => {
  const formData = new FormData()
  formData.append('file', file)
  return request.post<any, { url: string }>('/resume/upload-attachment', formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

/** PATCH /resume/publish 发布简历到人才市场 */
export const publishResume = () =>
  request.patch<any, void>('/resume/publish')

/** PATCH /resume/unpublish 从人才市场下架简历 */
export const unpublishResume = () =>
  request.patch<any, void>('/resume/unpublish')

/** GET /resume/search 搜索已发布的简历 */
export const searchPublishedResumes = (keyword?: string) =>
  request.get<any, ResumeData[]>('/resume/search', { params: { keyword } })

File: uniseek_java/src/main/java/com/uniseek/entity/Resume.java
package com.uniseek.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 简历实体
 */
@Data
@TableName("resume")
public class Resume {

    /** 简历 ID */
    @TableId
    private Long id;

    /** 用户 ID */
    private Long userId;

    /** 性别：0 男 / 1 女 */
    private Integer gender;

    /** 出生日期 */
    private LocalDate birthDate;

    /** 最高学历 */
    private String education;

    /** 毕业院校 */
    private String school;

    /** 技能特长 */
    private String skills;

    /** 工作经历 */
    private String experience;

    /** 附件简历 URL */
    private String attachmentUrl;

    /** 是否已发布到人才市场：0 未发布 / 1 已发布 */
    private Integer isPublished;

    /** 真实姓名（来自 real_name_auth 表 JOIN 查询，非数据库字段） */
    @TableField(exist = false)
    private String realName;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 更新时间 */
    private LocalDateTime updateTime;
}